projects/node-http-exercise/node-http-exercise.md

```diff
@@ -0,0 +1,255 @@
+    // if this route has a custom response, use it
+    if (routeResponse.response) {
+      response = routeResponse.response;
+    }
+
+    // if this route has custom headers, populate them for the response
+    if (routeResponse.header) {
+      header = { ...routeResponse.header };
+    }
+  }
+}
+```
+
+When testing this by going to `http://localhost:5000/test`, you should get:
+
+```text
+{"message":"It Works!"}
+```
+
+### Step 3 - Create Your Routes
+
+This is the core step of the assignment. Since you have a working structre, the endpoints creation should be easier.
+
+Now, we will build the following routes requested in your API. If you get stuck and need some help in creating the routes, you can look in the [hints](#hints) section for help.
+
+1. Create the `/spin` route, wich should make a GET request to the server and this one should respond with nothing, but keep on being open for connections.
+
+2. Create the `/hangup` route, which should close the connection with no response. This one can be created in multiple ways. Think about maybe making a check for this route before the helper function. Remember that you are looking for the `pathname` for this check. See [hints](#hints) for more information about this.
+
+3. Create the `/head` route, which should should respond with 204.
+
+4. Create the `/hello` route, which should should respond with 200 and text body of "Hello".
+
+5. Create the `/echo?message=xyz` route, which should respond with 200 and the message as the body. For this one, to help you out, you should start by console-logging the `req` object to find out how to pull out the message. Also, as a hint, the solution uses a check before the helper function also for that one.
+
+6. Create the `/error` route, which should should respond with 500.
+
+7. Create the `/time.json` route, which should should respond with 200, body with Content-Type of `application/json` and `{ "now": "Sat Feb 15 2020 14:29:20 GMT-0800" }` where it's the current time of the response.
+
+8. Create the `/github` route, which should respond with 302 that redirects to [Techtonica GitHub page](https://github.com/Techtonica).
+
+### Questions to Consider
+
+- Why do you think it is better practice not to hard-code everything within the response object and header?
+- How is this API construct different than a RESTful API?
+- In terms of security, what do you think the implementation with the different checks for routes either in the object or earlier certain endpoints yield? Could this be improved?
+
+### Hints
+
+- You can abruptly end a connection by using `res.end()` and `server.close()`.
+- If called as such, the `Date()` created will look like this: `Wed Oct 14 2020 21:22:50 GMT-0400 (Eastern Daylight Time)`, but for this exercise, we do not want what's in the parentheses at the end. For any more information, I invite you to check the [official Date documentation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date).
```

Comment:
The exact date format is locale dependent and we don't really care about it so I'd just remove this hint. Even if they output `Date.now()` and just the unix timestamp it would be successful in my opinion.

Reply:
fixed in [799330c](https://github.com/Techtonica/curriculum/pull/1406/commits/799330cf2b82a0e0d74d3cd2e258f97adc3e70db)